libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -501,6 +512,65 @@ void AP_Camera::take_picture()
     GCS_MAVLINK::send_to_components(MAVLINK_MSG_ID_COMMAND_LONG, (char*)&cmd_msg, sizeof(cmd_msg));
 }
 
+// start/stop recording video
+// start_recording should be true to start recording, false to stop recording
+void AP_Camera::record_video(bool start_recording)
+{
+#if HAL_MOUNT_ENABLED
+    // only mount implements recording video
+    if (get_trigger_type() == CamTrigType::mount) {
+        AP_Mount* mount = AP::mount();
+        if (mount != nullptr) {
+            mount->record_video(0, start_recording);
+        }
+    }
+#endif
+}
+
+// zoom in, out or hold
+// zoom out = -1, hold = 0, zoom in = 1
+void AP_Camera::set_zoom_step(int8_t zoom_step)
+{
+#if HAL_MOUNT_ENABLED
+    // only mount implements set_zoom_step
+    if (get_trigger_type() == CamTrigType::mount) {
+        AP_Mount* mount = AP::mount();
+        if (mount != nullptr) {
+            mount->set_zoom_step(0, zoom_step);
+        }
+    }
+#endif
+}
+
+// focus in, out or hold
+// focus in = -1, focus hold = 0, focus out = 1
+void AP_Camera::set_manual_focus_step(int8_t focus_step)
```

Comment:
These should all be boolean; the triggers via do_aux_function do have the capacity to return false, in which case the user will get an error message if they trigger them via mavlink, and the AUXF log message will get failure logged.

Reply:
ok, changed to be all booleans.